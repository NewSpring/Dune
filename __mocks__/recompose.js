// This should be able to be called from tests but currently
// it is not working.

// import { curry, fromPairs, map, adjust, toPairs } from "ramda";
// import R from "recompose";
//
// const mapValues = curry((fn, obj) =>
//   fromPairs(map(adjust(fn, 1), toPairs(obj))),
// );
//
// const withJest = mapValues(fn => jest.fn(fn));
//
console.log("hai");
// export default withJest(R);
