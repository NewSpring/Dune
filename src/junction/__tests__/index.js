// @flow
import { Text, View, Platform, Button } from "react-native";
import { withState, withHandlers } from "recompose";
import React from "react";
import renderer from "react-test-renderer";

import Junction from "~/junction";

const state = withState(
  "counter",
  "setCounter",
  ({ defaultValue }) => defaultValue || 0,
);

const actions = withHandlers({
  increment: ({ setCounter }) => () => setCounter(n => n + 1),
  decrement: ({ setCounter }) => () => setCounter(n => n - 1),
});

const Counter = ({
  increment,
  counter,
  decrement,
}: {
  increment: Function,
  counter: number,
  decrement: Function,
}) => (
  <View>
    <Text>Counter is at {counter} on {Platform.OS}</Text>
    <Button onPress={increment} title="Increment" />
    <Button onPress={decrement} title="Decrement" />
  </View>
);

const Item = Junction().with(state).with(actions).render(Counter);

describe("constructor", () => {
  it("should return an instance of Junction", () => {
    const junc = Junction();
    expect(junc instanceof Junction);
  });
});

describe("render", () => {
  const A = Junction().render(() => <Text>Hello World</Text>);
  const tree = renderer.create(<A />);
  expect(tree).toMatchSnapshot();
});

describe("map", () => {
  const A = Junction().map(hoc => B => hoc(() => <View><B /> </View>));

  const Comp = () => <Text>Hai World</Text>;

  const B = A.render(Comp);
  const C = A.render(() => null);

  it("renders a wrapped component", () => {
    const tree1 = renderer.create(<B />);
    expect(tree1).toMatchSnapshot();
  });
  it("renders null if passed", () => {
    const tree2 = renderer.create(<C />);
    expect(tree2).toMatchSnapshot();
  });
});

describe("with", () => {
  it("successfully wraps a component", () => {
    const A = Junction()
      .with(state)
      .with(actions)
      .render(({ foo, counter, increment }) => (
        <Text>{counter} - {foo} - {typeof increment}</Text>
      ));

    const tree = renderer.create(<A foo="bar" />);
    expect(tree).toMatchSnapshot();
  });
});

describe("Counter", () => {
  it("full integration counter", () => {
    // XXX test actual actions (i.e. click)
    const tree = renderer.create(<Item />);
    expect(tree).toMatchSnapshot();
  });
});
