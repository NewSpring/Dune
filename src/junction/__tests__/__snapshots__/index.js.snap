// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Hello World
</Text>
`;

exports[`Counter full integration counter 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Counter is at 
    0
     on 
    ios
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increment
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Decrement
      </Text>
    </View>
  </View>
</View>
`;

exports[`map renders a wrapped component 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hai World
  </Text>
   
</View>
`;

exports[`map renders null if passed 1`] = `
<View>
   
</View>
`;

exports[`with successfully wraps a component 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  0
   - 
  bar
   - 
  function
</Text>
`;
